src/recognizer/offline_recognizer.rs

```diff
@@ -0,0 +1,272 @@
+}
+
+/// Configuration for offline/non-streaming NeMo CTC models.
+///
+/// Please refer to
+/// https://k2-fsa.github.io/sherpa/onnx/pretrained_models/offline-ctc/index.html
+/// to download pre-trained models
+pub struct OfflineNemoEncDecCtcModelConfig {
+    pub model: String, // Path to the model, e.g., model.onnx or model.int8.onnx
+}
+
+pub struct OfflineWhisperModelConfig {
+    pub encoder: String,
+    pub decoder: String,
+    pub language: String,
+    pub task: String,
+    pub tail_paddings: i32,
+}
+
+pub struct OfflineTdnnModelConfig {
+    pub model: String,
+}
+
+pub struct OfflineSenseVoiceModelConfig {
+    pub model: String,
+    pub language: String,
+    pub use_inverse_text_normalization: i32,
+}
+
+/// Configuration for offline LM.
+pub struct OfflineLMConfig {
+    pub model: String, // Path to the model
+    pub scale: f32,    // scale for LM score
+}
+
+pub struct OfflineModelConfig {
+    pub transducer: OfflineTransducerModelConfig,
+    pub paraformer: OfflineParaformerModelConfig,
+    pub nemo_ctc: OfflineNemoEncDecCtcModelConfig,
+    pub whisper: OfflineWhisperModelConfig,
+    pub tdnn: OfflineTdnnModelConfig,
+    pub sense_voice: OfflineSenseVoiceModelConfig,
+    pub tokens: String, // Path to tokens.txt
+
+    // Number of threads to use for neural network computation
+    pub num_threads: i32,
+
+    // 1 to print model meta information while loading
+    pub debug: i32,
+
+    // Optional. Valid values: cpu, cuda, coreml
+    pub provider: Option<String>,
+
+    // Optional. Specify it for faster model initialization.
+    pub model_type: Option<String>,
+
+    pub modeling_unit: Option<String>, // Optional. cjkchar, bpe, cjkchar+bpe
+    pub bpe_vocab: Option<String>,     // Optional.
+    pub tele_speech_ctc: Option<String>, // Optional.
+}
+
+/// Configuration for the offline/non-streaming recognizer.
+pub struct OfflineRecognizerConfig {
+    pub feat_config: FeatureConfig,
+    pub model_config: OfflineModelConfig,
+    pub lm_config: OfflineLMConfig,
+
+    // Valid decoding method: greedy_search, modified_beam_search
+    pub decoding_method: String,
+
+    // Used only when DecodingMethod is modified_beam_search.
+    pub max_active_paths: i32,
+    pub hotwords_file: String,
+    pub hotwords_score: f32,
+    pub blank_penalty: f32,
+    pub rule_fsts: String,
+    pub rule_fars: String,
+}
+
+/// It wraps a pointer from C
+pub struct OfflineRecognizer {
+    pub(crate) pointer: *const SherpaOnnxOfflineRecognizer,
+}
+
+/// It contains recognition result of an offline stream.
+pub struct OfflineRecognizerResult {
+    pub text: String,
+    pub tokens: Vec<String>,
+    pub timestamps: Vec<f32>,
+    pub lang: String,
+    pub emotion: String,
+    pub event: String,
+}
+
+impl Drop for OfflineRecognizer {
+    fn drop(&mut self) {
+        self.delete()
+    }
+}
+
+impl OfflineRecognizer {
+    /// Frees the internal pointer of the recognition to avoid memory leak.
+    fn delete(&mut self) {
```

Comment:
Remove. You can free it directly in impl drop

Reply:
This is a private method, and it is currently only used for drop.

Reply:
You can remove the private delete methods and do the logic directly in drop

Reply:
@thewh1teagle I'd recommend keeping it as you might use this private method when writing unit tests.

Reply:
If we'll need the logic for unit tests we simply can call drop(...) that will call the drop implementation.